### PR TITLE
add check that `keep` in `filter()` must be boolean

### LINKED ISSUE #253: `filter()` assumes non-starred `dict` as `keep` kwarg
# Observed (user) error

Just ran into a stupid error where I used `df.filter(dict(foo='bar')` instead of `df.filter(**dict(foo='bar')`, so rather than applying the expected `foo='bar'` rule, the function interprets the dict as the `keep` kwarg.

# Suggested solution

Add a check that `keep` is a boolean, raise error otherwise.

### PR DESCRIPTION
**Pull request by danielhuppmann**
# Please confirm that this PR has done the following:

- [x] Tests Added
- [x] Documentation Added
- [x] Description in RELEASE_NOTES.md Added

# Description of PR

A non-starred `dict` in `df.filter()` is mis-interpreted as the `keep` kwarg, returning a copy of the full (rather than downselected) `IamDataFrame`. This PR adds a check the `keep` must be boolean.

closes #253

### CODE REVIEW

**Comment by gidden** [comment]
Looks good @danielhuppmann. One request: can you add an explicit test for a non-bool'd `keep`? I think the last test is actually doing this, but it would be nice to have something explicit (e.g., `keep='foo'`).

File: tests/test_core.py [hunk 1]
```diff
@@ -201,8 +201,15 @@ def test_variable_depth_raises(test_df):
...
-def test_filter_error(test_df):
-    pytest.raises(ValueError, test_df.filter, foo='foo')
+def test_filter_error_illegal_column(test_df):
+    # filtering by column `foo` is not valid
+    pytest.raises(ValueError, test_df.filter, foo='test')
+
+
+def test_filter_error_keep(test_df):
+    # string or non-starred dict was mis-interpreted as `keep` kwarg, see #253
+    pytest.raises(ValueError, test_df.filter, model='foo')
```
Comment (gidden) [inline tests/test_core.py]: I think this needs to be: `pytest.raises(ValueError, test_df.filter, model='foo', keep=1)` or something similar to explicitly test the boolean-ness.

**Comment by gidden** [comment]
ok, lgtm! will merge when CI passes.

**Comment by danielhuppmann** [comment]
thanks @gidden for the review!

**Pull request merged**